comparative_benchmark/tf_xla/run_benchmarks.py

```diff
@@ -0,0 +1,162 @@
+    0, str(pathlib.Path(__file__).parents[2] / "comparative_benchmark"))
+
+from openxla.benchmark import def_types
+from openxla.benchmark.comparative_suite.tf import benchmark_definitions
+from openxla.benchmark.models import model_interfaces
+import benchmark_lib, utils
+
+_HLO_DUMP_DIR = "/tmp/hlo_dump"
+_TF_CPU_DEVICE = "/CPU:0"
+_TF_GPU_DEVICE = "/GPU:0"
+
+
+def bytes_to_mb(bytes: Optional[int]) -> Optional[float]:
+  return None if bytes is None else bytes / 1e6
+
+
+def _run_framework_benchmark(
+    model: def_types.Model,
+    input_npys: Sequence[pathlib.Path],
+    expect_npys: Sequence[pathlib.Path],
+    verify_params: Dict[str, Any],
+    warmup_iterations: int,
+    benchmark_iterations: int,
+    backend: str,
+) -> Dict[str, Any]:
+  tf_device = _TF_GPU_DEVICE if backend == "gpu" else _TF_CPU_DEVICE
+
+  try:
+    with tf.device(tf_device):
+      if tf_device == _TF_GPU_DEVICE:
+        tf.config.experimental.reset_memory_stats(tf_device)
+
+      model_module = importlib.import_module(model.model_impl.module_path)
+      model_obj: model_interfaces.InferenceModel = model_module.create_model(
+          **model.model_parameters)
+
+      inputs = [np.load(path) for path in input_npys]
+      expects = [np.load(path) for path in expect_npys]
+
+      # Run warmup.
+      warmup_latencies = []
+      for i in range(warmup_iterations):
+        start = time.perf_counter()
+        outputs = model_obj.forward(inputs)
+        tf.test.experimental.sync_devices()
+        end = time.perf_counter()
+        warmup_latencies.append(1000 * (end - start))
+
+      # Run benchmark.
+      latencies = []
+      last_outputs = None
+      for i in range(benchmark_iterations):
+        start = time.perf_counter()
+        last_outputs = model_obj.forward(inputs)
+        tf.test.experimental.sync_devices()
+        end = time.perf_counter()
+        latencies.append(1000 * (end - start))
+
+      if last_outputs is None:
+        raise ValueError("No benchmark runs.")
+
+      verdicts = utils.compare_tensors(outputs=last_outputs,
+                                       expects=expects,
+                                       **verify_params)
+      all_equal = True
+      for idx, verdict in enumerate(verdicts):
+        is_equal, max_diff = verdict
+        if not is_equal:
+          print(f"Output {idx} exceeds tolerance. Max diff: {max_diff}")
+          all_equal = False
+
+      if not all_equal:
+        raise ValueError("Output verification failed.")
+
+      # Retrieve memory stats.
+      if tf_device == _TF_GPU_DEVICE:
+        memory_info = tf.config.experimental.get_memory_info(tf_device)
+        device_peak_b = memory_info["peak"]
+      else:
+        # tf.config.experimental does not currently support measuring CPU memory usage.
+        device_peak_b = None
+      device_peak_mb = bytes_to_mb(device_peak_b)
+
+      # Roughly calculate compile time.
+      compile_time_s = None if not warmup_latencies else (
```

Comment:
Should we report in ms? (as what we are doing in JAX benchmarks)

Reply:
Changed to ms.